### PR TITLE
Surveys page show first survey question 345

### LINKED ISSUE #345: After clicking `answer button` on survey card at surveys page show first survey question
1. visit https://espoo.herokuapp.com/surveys
1. click answer button for some survey
1. should be redirected to first question of the survey

### PR DESCRIPTION
**Pull request by garciajordy**
fix #345 

# Question Endpoint

When clicking on the `Answer` button on a survey it was creating an `AnswersSurvey` but it wasn't redirecting the user to the first question.

In order to fix this, I added a questions controller with the show method and views in order to be redirected to the first question.

#### Only select the appropriate.

- [ ] **Model testing.**
- [ ] **Request testing.**
- [x] **System testing.**
- [ ] **No tests required.**

### CODE REVIEW

**Review by edimossilva — COMMENTED** [review]
@garciajordy , I think you made some mistake here, can you double check, please?

File: backend/spec/system/front_end/answers_survey_views_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      end
+
+      it 'redirects to the question page' do
+        expect(page).to have_current_path question_path(first_question)
+      end
+    end
+
+    context 'when not valid' do
+      before do
+        allow(AnswersSurvey).to receive(:create!).and_raise(StandardError)
```
Comment (garciajordy) [inline backend/spec/system/front_end/answers_survey_views_spec.rb:27]: 🔝

File: backend/spec/system/front_end/answers_survey_views_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        sign_in user
+        visit surveys_path
+        click_on 'Answer'
+      end
+
+      it 'redirects to the surveys page' do
```
Comment (edimossilva) [inline backend/spec/system/front_end/answers_survey_views_spec.rb:33]: you can add a test chase to check if the message from `error: err.message` is visible. Because if we are adding this message here we should be seeing it in somewhere

File: backend/spec/system/front_end/question_views_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'Question', type: :system do
+  let!(:user) { create(:user_student) }
+  let!(:question) { create(:multiple_choice_question) }
+
+  describe '#show' do
+    before do
+      sign_in user
+      visit question_path question
```
Comment (edimossilva) [inline backend/spec/system/front_end/question_views_spec.rb:10]: I think that is not part of this PR but should we display only questions in some cases, please create an issue for this

File: backend/spec/system/front_end/question_views_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    it 'displays the name of the first question' do
```
Comment (edimossilva) [inline backend/spec/system/front_end/question_views_spec.rb:13]: why first?

File: backend/app/controllers/answers_surveys_controller.rb [hunk 5]
```diff
@@ -1,10 +1,19 @@
 class AnswersSurveysController < ApplicationController
   def create
```
Comment (edimossilva) [inline backend/app/controllers/answers_surveys_controller.rb:2]: what about this?
```
answers_survey = AnswersSurvey.create!(create_answers_survey_params)
redirect_to answers_survey.survey.questions.first
```

File: backend/app/views/questions/show.slim [hunk 6]
```diff
@@ -0,0 +1,32 @@
+h1
+  = @question.name
+h5
+  | This is a 
+  = @question.question_type.name.downcase
+  |  question.
+ul
+  - @question.options.each do |option|
+    - if @question.single_choice?
+      |  <input type='radio' id="
```
Comment (edimossilva) [inline backend/app/views/questions/show.slim:10]: ops, looks like there are a few thing incorrect in this syntax, please doublecheck this file

File: backend/app/views/questions/show.slim [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      | " value="
+      = option.name
+      | "> <label for="
+      =option.name
+      | "> 
+      = option.name
+    - else
+      input[type="text"]
+button
+  | Next
```
Comment (edimossilva) [inline backend/app/views/questions/show.slim:32]: empty line, maybe there is a rule for this on rubocop or in some other place

File: backend/spec/system/front_end/answers_survey_views_spec.rb (hunk 1 above)
Comment (edimossilva) [reply backend/spec/system/front_end/answers_survey_views_spec.rb:27]: top!

**Review by edimossilva — CHANGES_REQUESTED** [review]
Please, check a few things

good work btw, I think from now the project complexity will increase

File: backend/spec/system/front_end/question_views_spec.rb (hunk 3 above)
Comment (garciajordy) [reply backend/spec/system/front_end/question_views_spec.rb:10]: It should show the first question, right?

File: backend/app/views/questions/show.slim (hunk 7 above)
Comment (garciajordy) [reply backend/app/views/questions/show.slim:32]: I think rubocop doesn't check front end files, maybe we could look for another gem that can help with that?

File: backend/app/views/questions/show.slim (hunk 6 above)
Comment (garciajordy) [reply backend/app/views/questions/show.slim:10]: Hmm 🤔

File: backend/app/controllers/answers_surveys_controller.rb (hunk 5 above)
Comment (garciajordy) [reply backend/app/controllers/answers_surveys_controller.rb:2]: https://github.com/espoo-dev/espoo-dev/pull/403/commits/89179ad0a306f1283c0d938383c8b058ebb3625e

File: backend/spec/system/front_end/answers_survey_views_spec.rb (hunk 2 above)
Comment (garciajordy) [reply backend/spec/system/front_end/answers_survey_views_spec.rb:33]: https://github.com/espoo-dev/espoo-dev/pull/403/commits/b795bac410ead7680bab8e1fca7d12ae1f00fbb2

File: backend/spec/system/front_end/question_views_spec.rb (hunk 4 above)
Comment (garciajordy) [reply backend/spec/system/front_end/question_views_spec.rb]: https://github.com/espoo-dev/espoo-dev/pull/403/commits/11043748550a219a493bd3bfe9235ff07b777bd4

File: backend/app/views/questions/show.slim (hunk 7 above)
Comment (garciajordy) [reply backend/app/views/questions/show.slim]: https://github.com/espoo-dev/espoo-dev/pull/403/commits/324b55f0ba079997f70fc2e91067637d18887620

File: backend/app/views/questions/show.slim [hunk 8]
```diff
@@ -0,0 +1,32 @@
+h1
+  = @question.name
+h5
+  | This is a
```
Comment (edimossilva) [inline backend/app/views/questions/show.slim:4]: Ops, just realized that we should add translations here as well.
we can also let it go as this and create an issue for it

**Review by edimossilva — CHANGES_REQUESTED** [review]
